iothub/device/src/Authentication/ClientAuthenticationWithRegistrySymmetricKey.cs

```diff
@@ -87,25 +87,12 @@ public string ModuleId
         }
 
         /// <summary>
-        /// Gets or sets the key associated with the device.
+        /// Gets or sets the Base64 formatted shared access key associated with the device.
         /// </summary>
-        [SuppressMessage(
-            "Performance",
-            "CA1819:Properties should not return arrays",
-            Justification = "Cannot change property types on public classes.")]
-        public byte[] Key
+        public string Key
         {
-            get => _key;
-            set => _key = value ?? throw new InvalidOperationException("Shared access key cannot be null.");
-        }
-
-        /// <summary>
-        /// Gets or sets the Base64 formatted key associated with the device.
-        /// </summary>
-        public string KeyAsBase64String
-        {
-            get => Convert.ToBase64String(Key);
-            set => SetKeyFromBase64String(value);
+            get => Convert.ToBase64String(_key);
```

Comment:
Do we even need to save the key privately as a byte[]? I imagine we could just take a string as is and not need to convert to/from base64 here